go-controller/pkg/node/gateway_udn.go

```diff
@@ -456,7 +460,7 @@ func (udng *UserDefinedNetworkGateway) DelNetwork() error {
 	}
 
 	if util.IsPodNetworkAdvertisedAtNode(udng.NetInfo, udng.node.Name) {
-		err := udng.updateAdvertisedUDNIsolationRules(false)
+		err := udng.deleteAdvertisedUDNIsolationRules()
```

Comment:
OK, this opens a new can of worms :P Bear with me.

I think this delete call should be unconditional and ensure there's no network leftover.

Reply:
I will leave it up to you, but `deleteAdvertisedUDNIsolationRules` is (and should be) idempotent so unconditional call is definitely the safest approach if we are actually deleting the network.